conbench/entities/run.py

```diff
@@ -101,57 +103,93 @@ def create(data):
         run.save()
         return run
 
-    def get_baseline_run(self):
+    def get_baseline_run(
+        self,
+        commit_limit: int = 20,
```

Comment:
We can leave this as is for now, but a few ideas that might make this restriction not so bad (none of these are mutually exclusive, so we might end up doing all of them):

* Could we have the code below iterate through in chunks of 20 until it finds history? This way the bast majority of cases (where within 20 commits, there's _something_ that could be a baseline) is fast, but if that's not found we degrade gracefully (e.g. find something even if it takes 5x as long going back 100 commits) 

* Have you profiled to see what makes this query take so long? Maybe it's possible to pass a subset of the data to the ancestor query (only the benchmarks we're interested in) that would make walking back like this faster

* Should we make this tuneable (like we did for the history distribution look back window)? In case there are repos where large gaps happen + they don't mind the slow performance?

Reply:
Good ideas! Some thoughts on them:
- Re: iteration, I suspect it's more common for a run to be the "true" first run in history (e.g. the first with a certain hardware or new set of case/contexts) than for it to have a baseline that's more than 20 commits away. But I don't have the database stats to back this up. 🙂 either situation is probably pretty rare! But yeah maybe we could exponentially iterate at most 3 times so that "true" first runs don't get stuck. We also mustn't take more than 30 seconds or we'll return an Internal Server Error.
- Yes, I profiled the python code before I introduced the `commit_limit`, and 99%+ of the time is spent waiting for the database to execute the query. So it's definitely a query optimization problem. I used EXPLAIN ANALYZE to look at the different parts of the execution, and while I'm fairly inexperienced at doing that, I could tell that most of the time was spent evaluating whether runs on very old commits matched the criteria or not. That's why `commit_limit` worked so well.
- Yep this is a great idea.

Reply:
Made #569 and #570.